File: PkmWebServer/Models/States/GameStateModel.cs
using PkmWebServer.Dtos.Game;
using PkmWebServer.Models.Refs;

namespace PkmWebServer.Models.States
{
    public class GameStateModel
    {
        public PkmRefModel Ref { get; set; } = new();
        public ISet<string> Guesses { get; set; } = new HashSet<string>();
        public ICollection<string> WrongGuesses { get; set; } = [];
        public int Lives { get; set; } = 0;
        public IDictionary<string, int> MoveNameKeys { get; set; } = new Dictionary<string, int>();
        public IDictionary<int, MoveStateModel> MoveStates { get; set; } = new Dictionary<int, MoveStateModel>();
        public GameResultTypes Result { get; set; } = GameResultTypes.Ongoing;
        public StatsModel Stats { get; set; } = new();
    }
}

File: PkmWebClient/src/app/guesser/game/proc/game-state.model.ts
import { GameResultTypes } from "@guesser/game/models";
import { MoveState } from "@guesser/guess/models";

export interface GameState {
  name: string,
  type1: string,
  type2: string | null,
  moves: MoveState[],
  wrongGuesses: string[],
  lives: number,
  result: GameResultTypes
}
